subsys/bluetooth/host/mesh/beacon.c

```diff
@@ -302,11 +303,18 @@ static void secure_beacon_recv(struct net_buf_simple *buf)
 		bt_mesh_net_beacon_update(sub);
 	}
 
-	if (IS_ENABLED(CONFIG_BT_MESH_FRIEND) && (iv_change || kr_change)) {
-		if (iv_change) {
-			bt_mesh_friend_sec_update(BT_MESH_KEY_ANY);
-		} else {
-			bt_mesh_friend_sec_update(sub->net_idx);
+	if (iv_change || kr_change) {
+		if (IS_ENABLED(CONFIG_BT_MESH_FRIEND)) {
+			if (iv_change) {
+				bt_mesh_friend_sec_update(BT_MESH_KEY_ANY);
+			} else {
+				bt_mesh_friend_sec_update(sub->net_idx);
+			}
+		}
+
+		if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY) &&
+		    bt_mesh_gatt_proxy_get() == BT_MESH_GATT_PROXY_ENABLED) {
+			bt_mesh_proxy_beacon_send(sub);
```

Comment:
As with the Friend update, an IV Update affects all subnets (even though the beacon is only authenticated for one), so I'll fine-tune this patch a bit so that we have similar functionality as for the Friend update.

Reply:
FWIW, this doesn't inpact PTS testing since the PTS never tests more than one subnet when it comes to these test cases.